addons/web/static/src/model/relational_model/record.js

```diff
@@ -703,7 +695,9 @@ export class Record extends DataPoint {
                 this.fields[propertyFieldName] = {
                     ...property,
                     name: propertyFieldName,
-                    relatedPropertyField,
+                    relatedPropertyField: {
+                        name: fieldName,
```

Comment:
why rename fieldName => name ?

Reply:
Here, we add the property to the `fields` object which is shared with other components of the view.
In the search model, we also add properties to the same object (named `searchViewFields` in the search model) but we the data are incompatible. In the search model `relatedPropertyField` is a reference to a `fields` entry with the structure of a field definition (so we use `name` and not `fieldName`).
I only give `name` here in the relational model because it's the minimum we need. I set the `parent` info in the related active field to not have model data in the field definitions object.